client-v2/src/shared/bundles/collectionsBundle.js

```diff
@@ -23,101 +16,35 @@ const collectionSelectors = {
   ) => {
     let collectionId = null;
     Object.keys(state.collections.data).some(id =>
-      Object.keys(state.collections.data[id].articleFragments).some(stage => {
-        if (!state.collections.data[id].articleFragments[stage]) {
-          return false;
-        }
-        // If we've got a hit here, stop the search ...
-        if (
-          state.collections.data[id].articleFragments[stage] &&
-          state.collections.data[id].articleFragments[stage].indexOf(
-            articleFragmentId
-          ) !== -1
-        ) {
-          collectionId = id;
-          return true;
-        }
-        // ... else, inspect the supporting articles, if they exist.
-        state.collections.data[id].articleFragments[stage].some(fragmentId => {
-          if (
-            state.articleFragments[fragmentId] &&
-            state.articleFragments[fragmentId].meta &&
-            state.articleFragments[fragmentId].meta.supporting &&
-            state.articleFragments[fragmentId].meta.supporting.indexOf(
-              articleFragmentId
-            ) !== -1
-          ) {
+      ['live', 'draft', 'previously'].some(stage => {
+        const groups = state.collections.data[id][stage] || [];
+
+        return groups.some(gId => {
```

Comment:
Personally I'd call it `groupID` you're using `collectionId`

Reply:
You're definitely (cont ...)